### DISCUSSION TITLE
Question - Lettuce Lazy Connection Creation

### DISCUSSION THREAD

**Discussion by kandogu** [redis/lettuce#2247] [Q&A]
## Feature Request

I checked lettuce-wiki but I think I couldn't find an answer for my question, so I just wanted to ask from here. I really appreciated if we can make it clear.

#### Is your feature request related to a problem? Please describe

So our problem is lazy connection creation in redis, in the config we set min idle connection count but it doesn't apply our min idle config. We expected to see that all idle connection must be created on application startup. But, when we run our application, only required amount of connection created between redis and our application. However, it is a problem for our use case because in our app we mostly get temporary peaks and when we have peaks lettuce try to create lots of connection and it decreases application performance. Is it possible to open all idle connections on application startup with lettuce-redis?

By the way, we don't use shared connection.

Thank you

**Comment by mp911de** [comment]
Have you tried calling `GenericObjectPool.preparePool()`? (assuming you're using `        GenericObjectPool<StatefulRedisConnection<String, String>> pool = ConnectionPoolSupport.createGenericObjectPool(() -> client.connect(), new GenericObjectPoolConfig<>());`).

**Reply by kandogu** [reply]
Yes, I'm using `createGenericObjectPool()` but it is implemented in `getConnection()` function of `LettucePoolingConnectionProvider` class, so it is a class of `spring-data-redis`, I guess issue related to spring framework then, because I cannot access `genericObjectPool` to call `preparePool()` function.

**Reply by mp911de** [reply]
Wait, `LettucePoolingConnectionProvider` is Spring Data Redis. Can you file a ticket at https://github.com/spring-projects/spring-data-redis/issues, please?

**Reply by kandogu** [reply]
Okey, I'll open this issue there, thank you
